src/container/srv_target.c

```diff
@@ -1196,7 +1206,9 @@ ds_cont_hdl_get(struct ds_cont_hdl *hdl)
 {
 	struct d_hash_table *hash = &dsm_tls_get()->dt_cont_hdl_hash;
 
-	cont_hdl_get_internal(hash, hdl);
+	/* It's global srv cont handle when the sch_cont == NULL */
+	if (hdl->sch_cont != NULL)
+		cont_hdl_get_internal(hash, hdl);
```

Comment:
Why not hold reference for the global srv cont handle? Just because it is not in hash table? If not hold the reference, why related sponsor need to call `ds_cont_hdl_get()` against the global srv cont handle?

Reply:
The global srv cont handle is per pool and immutable, so it's not necessary to hold reference for it.
In current implementation, the caller doesn't know if a handle is global handle or normal handle and uses same set of API for the container handle.

Reply:
But if we hold reference on that, it is harmful or not? If not, unconditional holding reference seems more simple.

Reply:
cont_hdl_get_internal() can't be called for server handle, the server handle isn't inserted in open handle hash.